src/Microsoft.DotNet.Interactive.CSharpProject/RoslynWorkspaceUtilities.cs

```diff
@@ -0,0 +1,389 @@
+    {
+        string projectName = Path.GetFileNameWithoutExtension(buildProjectData.ProjectFilePath);
+        if (!TryGetSupportedLanguageName(buildProjectData.ProjectFilePath, out string languageName))
+        {
+            return null;
+        }
+        return ProjectInfo.Create(
+            projectId,
+            VersionStamp.Create(),
+            projectName,
+            projectName,
+            languageName,
+            filePath: buildProjectData.ProjectFilePath,
+            outputFilePath: buildProjectData.PropertyTargetPath,
+            documents: GetDocuments(buildProjectData, projectId),
+            projectReferences: GetExistingProjectReferences(buildProjectData, workspace),
+            metadataReferences: GetMetadataReferences(buildProjectData),
+            analyzerReferences: GetAnalyzerReferences(buildProjectData, workspace),
+            parseOptions: CreateParseOptions(buildProjectData, languageName),
+            compilationOptions: CreateCompilationOptions(buildProjectData, languageName));
+    }
+
+    private static bool TryGetSupportedLanguageName(string projectPath, out string languageName)
+    {
+        switch (Path.GetExtension(projectPath))
+        {
+            case ".csproj":
+                languageName = LanguageNames.CSharp;
+                return true;
+            default:
+                languageName = null;
+                return false;
+        }
+    }
+
+    private static IEnumerable<DocumentInfo> GetDocuments(BuildProjectData buildProjectData, ProjectId projectId) =>
+        buildProjectData
+            .SourceFiles?.Where(File.Exists)
+    .Select(x => DocumentInfo.Create(
+                DocumentId.CreateNewId(projectId),
+    Path.GetFileName(x),
+                loader: TextLoader.From(
+                    TextAndVersion.Create(
+                        SourceText.From(File.ReadAllText(x), Encoding.Unicode), VersionStamp.Create())),
+                filePath: x))
+        ?? Array.Empty<DocumentInfo>();
+
+    private static IEnumerable<ProjectReference> GetExistingProjectReferences(BuildProjectData buildProjectData, Microsoft.CodeAnalysis.Workspace workspace) =>
+        buildProjectData.ProjectReferences
+            .Select(x => workspace.CurrentSolution.Projects.FirstOrDefault(y => y.FilePath.Equals(x, StringComparison.OrdinalIgnoreCase)))
+
+            .Where(x => x != null)
+            .Select(x => new ProjectReference(x.Id))
+        ?? Array.Empty<ProjectReference>();
+
+    private static IEnumerable<MetadataReference> GetMetadataReferences(BuildProjectData buildProjectData) =>
+        buildProjectData
+            .References?.Where(File.Exists)
+            .Select(x => MetadataReference.CreateFromFile(x))
+        ?? (IEnumerable<MetadataReference>)Array.Empty<MetadataReference>();
+
+
+    private static IEnumerable<AnalyzerReference> GetAnalyzerReferences(BuildProjectData buildProjectData, Microsoft.CodeAnalysis.Workspace workspace)
```

Comment:
Ditto,

Reply:
Done.

Now this method returns IReadOnlyList